tests/contracts/basetests.ts

```diff
@@ -0,0 +1,164 @@
+import StellarSdk from 'stellar-sdk'
+import SorobanClient from 'soroban-client'
+import { spawn } from 'child_process'
+import type { AppConfig } from '../appconfig'
+
+export enum OperationType {
+  UploadContract,
+  CreateContract,
+  InvokeMethod,
+}
+
+export class OperationFee {
+  fee: number
+  operationType: OperationType
+  methodName: string
+}
+
+export abstract class BaseTests {
+  fees: Map<string, OperationFee[]> = new Map<string, OperationFee[]>()
+  deployFee: number = 0
+  invokeFee: number = 0
+  protected logs: string[] = new Array<string>()
+  protected accounts: Map<string, string[]> = new Map<string, string[]>([
+    ['admin', []],
+    ['player_1', []],
+    ['player_2', []],
+    ['player_3', []],
+  ])
+
+  protected contractId: string = ''
+
+  constructor(protected config: AppConfig) {}
+
+  async createAccounts() {
+    let resultOk = true
+    for (const acc of this.accounts.keys()) {
+      await this.runSoroban(['config', 'identity', 'generate', acc])
+      await this.runSoroban(['config', 'identity', 'address', acc]).then(
+        async data => {
+          const id = data.toString().trim()
+          await this.runSoroban(['config', 'identity', 'show', acc]).then(
+            pwd => {
+              this.accounts.set(acc, [id, pwd.toString().trim()])
+            }
+          )
+          const url = `${this.config.friendbot}?addr=${id}`
+          const res = await fetch(url)
+          const json = await res.json()
+
+          if (this.config.showLogs) {
+            console.log(json)
+          }
+        },
+        error => {
+          resultOk = false
```

Comment:
This continues executing even if the result is false. Why not use [Promises.all](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all)

Reply:
Actually, we may want to introspect when something fails, so we might prefer to use `Promises.allSettled`